packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,116 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+
+export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** If the link will redirect users to a page in or out Grafana */
+  externalLink?: boolean;
+  /** Specify the icon name to be used */
+  icon?: IconName;
+  /** To clarify if the link is a block of if it is part of a bigger text */
+  inline?: boolean;
+  /** What typograpy variant should be used for the component. Only use if default variant for the defined element is not what is needed */
+  variant?: keyof ThemeTypographyVariantTypes;
+  /** Override the default weight for the used variant */
+  weight?: 'light' | 'regular' | 'medium' | 'bold';
+  children: string;
+}
+
+/**
+ * @alpha
+ */
+export const TextLink = forwardRef<HTMLAnchorElement, Props>(
+  ({ href, color, externalLink = false, inline, variant, weight, icon, children, ...rest }, ref) => {
+    const validUrl = locationUtil.stripBaseFromUrl(textUtil.sanitizeUrl(href ?? ''));
+
+    const theme = useTheme2();
+    const styles = getLinkStyles(theme, variant, weight, color, inline);
+
+    return externalLink ? (
+      <RouterLink ref={ref} to={validUrl} {...rest} className={styles}>
+        {children}
+        {icon && !inline && <Icon name={icon} />}
+      </RouterLink>
+    ) : (
+      <a href={validUrl} target="_blank" rel="noreferrer" className={styles}>
+        {children}
+        {icon && !inline && <Icon name={icon} />}
```

Comment:
Discuss with Staton if we would want to show external icon inline as well, if not, mention it in the props docs as well